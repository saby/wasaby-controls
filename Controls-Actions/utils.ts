import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
