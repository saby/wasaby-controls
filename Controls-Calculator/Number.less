.calculator-View__number__clickable {
   cursor: pointer;
}
