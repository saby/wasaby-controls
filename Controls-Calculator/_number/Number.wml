<ws:partial scope="{{ _options }}"
            template="{{_options.content}}"
            on:mousedown="_clickHandler()"
            on:click="_clickHandler()"
            attr:class="controls_Calculator_theme-{{_options.theme}} calculator-View__number
            {{_isCalculatorRegistered ? 'calculator-View__number__clickable'}}"
/>
