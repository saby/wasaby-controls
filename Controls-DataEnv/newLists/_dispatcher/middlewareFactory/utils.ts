export const addName = <T extends Function>(label: string, fn: T): T => {
    const wrapper = {
        [label]: ((...args: unknown[]) => fn(...args)) as unknown as T,
    };

    return wrapper[label];
};

const POSTFIX = '{name}';
export const createMask = <T extends string>(prefix: T) => `${prefix}(${POSTFIX})` as const;
export const decorateName = (mask: string, name: string) => mask.replace(POSTFIX, name);
export const undecorateName = (mask: string, decoratedName: string) => {
    const [start, end] = mask.split(POSTFIX);
    return decoratedName.replace(start, '').replace(end, '');
};
