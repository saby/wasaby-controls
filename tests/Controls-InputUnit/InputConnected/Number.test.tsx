/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';
import { getContent } from '../resources/utils';
import { default as DefaultValue } from 'Controls-Input-demo/InputConnected/Number/DefaultValue';
import { default as Label } from 'Controls-Input-demo/InputConnected/Number/Label';
import { default as IntegersLength } from 'Controls-Input-demo/InputConnected/Number/IntegersLength';
import { default as OnlyPositive } from 'Controls-Input-demo/InputConnected/Number/OnlyPositive';
import { default as Precision } from 'Controls-Input-demo/InputConnected/Number/Precision';
import { default as UseGrouping } from 'Controls-Input-demo/InputConnected/Number/UseGrouping';

describe(
    'Controls-Input/InputConnected:Number',
    () => {
        setTestID(148321529);

        let container = null;

        beforeEach(() => {
            jest.useFakeTimers('modern');
            jest.setSystemTime(new Date(1996, 5, 21));
            container = document.createElement('div');
            WasabyEvents.initInstance(container);
            document.body.appendChild(container);
        });
        afterEach(() => {
            unmountComponentAtNode(container);
            WasabyEvents.destroyInstance(container);
            jest.useRealTimers();
            container.remove();
            container = null;
        });

        it('defaultValue', async () => {
            render(getContent(DefaultValue), {container});
            expect(container).toMatchSnapshot();
        });
        it('IntegersLength', async () => {
            render(getContent(IntegersLength), {container});
            expect(container).toMatchSnapshot();
        });
        it('label', async () => {
            render(getContent(Label), {container});
            expect(container).toMatchSnapshot();
        });
        it('OnlyPositive', async () => {
            render(getContent(OnlyPositive), {container});
            expect(container).toMatchSnapshot();
        });
        it('Precision', async () => {
            render(getContent(Precision), {container});
            expect(container).toMatchSnapshot();
        });
        it('UseGrouping', async () => {
            render(getContent(UseGrouping), {container});
            expect(container).toMatchSnapshot();
        });
    }
);
