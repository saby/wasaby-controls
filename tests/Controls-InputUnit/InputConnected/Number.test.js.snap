// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/InputConnected:Number IntegersLength 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            integersLength = 2
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            integersLength is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Number OnlyPositive 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            onlyPositive = false
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            onlyPositive = true
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Number Precision 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            precision = 2
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            precision is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Number UseGrouping 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            useGrouping = false
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            useGrouping = true
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Number defaultValue 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            defaultValue = 2
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="2"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            defaultValue is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
            />
            <div
              class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Number label 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth300 tw-flex tw-flex-col"
  >
    <div
      class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default controls-margin_top-m controls-margin_top-m controls-Input-connected_filled"
      tabindex="0"
    >
      <div
        class="controls-Render__wrapper"
      >
        <span
          class="controls-Render__baseline"
        >
          ﻿
        </span>
        <div
          class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
          data-qa="controls-Render__field"
          tabindex="0"
        >
          <input
            autocapitalize="off"
            autocomplete="off"
            autocorrect="off"
            class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
            inputmode="decimal"
            name="ws-input_1996-06-20"
            spellcheck="true"
            tabindex="0"
            type="text"
            value="22"
          />
          <div
            class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
          >
            <div
              class="controls-Render__placeholder controls-Render__placeholder_overflow"
            >
              Укажите число
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label top
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default controls-margin_top-m"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0 tw-flex-shrink"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label start
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default controls-input_label_min-input-content controls-margin_top-m"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
    >
      <span
        class="controls-icon_size-s controls-icon_style-secondary icon-SabyBird controls-icon controls-margin_right-s"
      />
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Number controls_Number_theme-default controls-margin_top-m"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="decimal"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="22"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Укажите число
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
